include/items.php

```diff
@@ -763,6 +764,17 @@ function item_store($arr,$force_parent = false, $notify = false, $dontcache = fa
 		return 0;
 	}
 
+	// Check for already added items.
+	// There is a timing issue here that sometimes creates double postings.
+	// An unique index would help - but the limitations of MySQL (maximum size of index values) prevent this.
```

Comment:
BINARY and VARBINARY aren't convenient for searching since it's exact match only, so it's not usable for all fields, but for GUIDs it is perfectly suitable.